src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1331,6 +1331,177 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
+                              GraphQL: new(Singular: "GetBooks", Plural: "GetBooks"),
+                              Permissions: new[] { GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                              Relationships: null,
+                              Mappings: null
+                             );
+
+                string entityName = "GetBooks";
+                configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions, entity, entityName);
+            }
+            else
+            {
+                configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions);
+            }
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+
+            AuthenticationOptions AuthenticationOptions = new(Provider: EasyAuthType.StaticWebApps.ToString(), null);
+            HostOptions staticWebAppsHostOptions = new(null, AuthenticationOptions);
+
+            RuntimeOptions runtimeOptions = configuration.Runtime;
+            RuntimeOptions baseRouteEnabledRuntimeOptions = new(runtimeOptions.Rest, runtimeOptions.GraphQL, staticWebAppsHostOptions, "/data-api");
+            RuntimeConfig baseRouteEnabledConfig = configuration with { Runtime = baseRouteEnabledRuntimeOptions };
+            File.WriteAllText(CUSTOM_CONFIG, baseRouteEnabledConfig.ToJson());
+
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                HttpMethod httpMethod = SqlTestHelper.ConvertRestMethodToHttpMethod(SupportedHttpVerb.Post);
+                HttpRequestMessage request = new(httpMethod, requestPath);
+                if (entityType is not EntitySourceType.StoredProcedure)
+                {
+                    string requestBody = @"{
+                        ""title"": ""Harry Potter and the Order of Phoenix"",
+                        ""publisher_id"": 1234
+                    }";
+
+                    JsonElement requestBodyElement = JsonDocument.Parse(requestBody).RootElement.Clone();
+                    request = new(httpMethod, requestPath)
+                    {
+                        Content = JsonContent.Create(requestBodyElement)
+                    };
+                }
+
+                HttpResponseMessage response = await client.SendAsync(request);
+                Assert.AreEqual(HttpStatusCode.Created, response.StatusCode);
+
+                string locationHeader = response.Headers.Location.AbsoluteUri;
+                Assert.IsTrue(locationHeader.StartsWith(expectedLocationHeader));
+
+                string path = response.Headers.Location.AbsolutePath;
+                string completeUrl = "http://localhost" + path.Substring(baseRoute.Length - 1);
```

Comment:
Can you please add the reason why completeUrl should skip baseRoute in code comments here?

Reply:
Yes, because when the request lands at DAB, the base-route would be stripped off from the url.

Reply:
`base-route` is used by SWA for redirection only. Any request to `/data-api` (or the configured base-route) gets redirected to phoenix clusters by SWA. However, all the APIs are hosted in DAB containers at the `/api`. 
So, when DAB is not integrated with SWA, to be able to perform a GET request, we need to remove the `base-route` from the URL.